### PR TITLE
Revert Dokka to 1.4.10.2; only generate docs for stable releases.

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### Summary of changes
This PR revert dokka to 1.4.10.2 so that the generated docs will keep the original structure.
Also update the CI config, now only the stable release can trigger the job of generating docs.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -178,10 +178,9 @@ commands:
       - run:
           name: Generate android docs and create PR for review.
           command: |
-            if [[ $CIRCLE_TAG == android-v* ]]; then
+            if [[ $CIRCLE_TAG == android-v* ]] && [[$CIRCLE_TAG != *alpha*]] && [[$CIRCLE_TAG != *beta*]] && [[$CIRCLE_TAG != *rc*]]; then
```
Comment (kiryldz) [inline .circleci/config.yml:181]: does it mean we won't generate API docs for alpha, beta and rc?
we should generated API docs but we should not bump version in android-docs repo.

Comment (Chaoba) [reply .circleci/config.yml:181]: I see, we should config it in script not here.

**Review by kiryldz — APPROVED** [review]
LGTM, great job @Chaoba!

File: buildSrc/src/main/kotlin/Project.kt [hunk 2]
```diff
@@ -71,7 +71,7 @@ object Versions {
   const val pluginAndroidGradle = "4.0.1"
   const val pluginKotlin = "1.4.0"
   const val pluginLicense = "0.8.5"
-  const val pluginDokka =  "1.4.32"
+  const val pluginDokka =  "1.4.10.2"
```
Comment (yunikkk) [inline buildSrc/src/main/kotlin/Project.kt:74]: BTW what's the issue with the updated Dokka? Is there a bug or structure is changed by intent? 
Anyway - would be nice to add a comment to not update to 1.4.32

**Review by pengdev — APPROVED** [review]

File: scripts/update-android-docs.sh [hunk 3]
```diff
@@ -135,15 +135,17 @@ create_pull_request "Add ${MAPS_SDK_VERSION} API documentation." $BRANCH_WITH_DO
...
-clone_android_docs_repo
-update_constants_and_map_version_numbers $MAPS_SDK_VERSION
-prepare_android_docs_branch $MAPS_SDK_VERSION
-cd $ANDROID_DOCS_DIRECTORY
-create_pull_request "Carbon Maps SDK bump to ${MAPS_SDK_VERSION}" $BRANCH_WITH_DOCUMENTATION
-# Rollback the remote url when run the script locally
-git remote set-url origin git@github.com:mapbox/android-docs.git
-cd -
+# Update config files in Android Docs Repo for stable release.
+if [[ $MAPS_SDK_VERSION != *beta* ]] && [[ $MAPS_SDK_VERSION != *alpha* ]] && [[ $MAPS_SDK_VERSION != *rc* ]]; then
```
Comment (alexander-kulikovskii) [inline scripts/update-android-docs.sh:139]:
```suggestion
if [[ $MAPS_SDK_VERSION !=~ ^(*beta*|*alpha*|*rc*)$ ]]; then
```
but check syntax

File: buildSrc/src/main/kotlin/Project.kt (hunk 2 above)
Comment (Chaoba) [reply buildSrc/src/main/kotlin/Project.kt:74]: The file structure would be changed with 1.4.32 and we can't deploy it. For more detail please refer to the linked ticekt.

File: scripts/update-android-docs.sh (hunk 3 above)
Comment (Chaoba) [reply scripts/update-android-docs.sh:139]: The script has been tested and worked well, so I would like to keep it.

File: buildSrc/src/main/kotlin/Project.kt (hunk 2 above)
Comment (ank27) [reply buildSrc/src/main/kotlin/Project.kt:74]: Yeah, I checked that there is a breaking change in the dokka task that breaks the structure in [v1.4.30 ](https://github.com/Kotlin/dokka/releases/tag/v1.4.30). we do use `dokkaHtml` to generate the source sets which is changed to `dokkaHtmlPartial`. that might be the reason why 1.4.32 doesn't work properly.

**Review by ank27 — APPROVED** [review]

File: buildSrc/src/main/kotlin/Project.kt (hunk 2 above)
Comment (Chaoba) [reply buildSrc/src/main/kotlin/Project.kt:74]: No, we will update it in the future, I have created to ticket for it.

Comment (yunikkk) [reply buildSrc/src/main/kotlin/Project.kt:74]: Ok then!

**Pull request merged**